auth0_flutter/test/web/auth0_fluter_web_test.dart

```diff
@@ -153,4 +153,15 @@ void main() {
 
     expect(() async => auth0.credentials(), throwsException);
   });
+
+  test('logout is called and succeeds', () async {
+    when(mockClientProxy.logout(any)).thenAnswer((final _) => Future.value());
+    await auth0.logout(federated: true, returnToUrl: 'http://returnto.url');
+
+    final params =
+        verify(mockClientProxy.logout(captureAny)).captured.first.logoutParams;
+
+    expect(params.federated, true);
+    expect(params.returnTo, 'http://returnto.url');
+  });
```

Comment:
Should we add a failing test? It'll have to be changed when we add our new exception, but at least it will be in place.

Reply:
Also, should we test that the nulls are stripped, like in "loginWithRedirect strips options that are null"?

Reply:
> Should we add a failing test?

I've added one now in [ab3524b](https://github.com/auth0/auth0-flutter/pull/223/commits/ab3524b9edb147e1daae8939a360b9ee285edd54) that can be rewritten later 👍🏻 

> Also, should we test that the nulls are being stripped, like in "loginWithRedirect strips options that are null"?

Yes I agree that's a good test, and in fact started down this road as part of the login work but decided to revisit it. There's no easy way to distinguish between something that's `null` and something that's not been set. If you take this example:

```dart
final params = verify(mockClientProxy.loginWithRedirect(captureAny))
    .captured
    .first
    .authorizationParams;

expect(params.audience, null);
```

`params.audience` will be `null` if it hasn't been set or if it's actually been set to `null`. To test `stripNulls` properly
we'll need to make use of [Dart's reflection capabilities](https://api.dart.dev/stable/2.19.4/dart-mirrors/dart-mirrors-library.html), the extent of which I haven't fully explored but presumably we'll be able to do some runtime dynamic trickery to figure out whether the property actually exists on `params`. I didn't want to block the login feature at the time but I'll add a separate ticket for this 👍🏻